docs/devel/e2e-tests.md

```diff
@@ -102,6 +102,18 @@ As mentioned earlier there are a host of other options that are available, but a
 - `rm -rf /var/run/kubernetes`, clear kube generated credentials, sometimes stale permissions can cause problems.
 - `sudo iptables -F`, clear ip tables rules left by the kube-proxy.
 
+## Kinds of tests
+
+We are working on implementing clearer partitioning of our e2e tests to make running a known set of tests easier (#10548).  Tests can be labeled with any of the following labels, in order of increasing precedence (that is, each label listed below supersedes the previous ones):
+
+- If a test has no labels, it is expected to run fast (under five minutes), be able to be run in parallel, and be consistent.
+- `[Slow]`: If a test takes more than five minutes to run (in parallel with many other tests), it is labeled `[Slow]`.  This partition allows us to run almost all of our tests quickly in parallel, without waiting for the stragglers to finish.
+- `[Serial]`: If a test cannot be run in parallel with other tests (e.g. it takes too many resources or restarts nodes), it is labeled `[Serial]`, and should be run in serial as part of a separate suite.
+- `[Disruptive]`: If a test restarts components that might cause other tests to fail, it is labeled `[Disruptive]`.  Any `[Disruptive]` test is also assumed to be `[Serial]`.  These tests are not run against soak clusters to avoid restarting components.
+- `[Flaky]`: If a test is found to be flaky, it receives the `[Flaky]` label until it is fixed.  A `[Flaky]` label should be accompanied with a reference to the issue for de-flaking the test.
+- `[Skipped]`: `[Skipped]` is a legacy label that we're phasing out.  If a test is marked `[Skipped]`, there should be an issue open to label it properly.  `[Skipped]` tests are never run.
```

Comment:
Actually these were used to cause a test to be skipped by default.  They are typically explicitly included in specific test jobs.  I guess you could force people to use one of the other labels instead (e.g. skip because it's slow, skip because it's disruprive, skip because it uses too many resources, skip because it's flaky etc) but it's been pretty useful to have a "skip for other reasons" label, which is essentially what this was used for.